### PR TITLE
Add apache kylin conn type and kylin connection docs

### PR DESCRIPTION
**Pull request by shohamy7**
---
Related: #28790
This PR adds connection type of Apache Kylin and add docs about the connection

**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by hussein-awala — APPROVED** [review]
Looks good, LGTM

File: airflow/providers/apache/kylin/hooks/kylin.py [hunk 1]
```diff
@@ -32,6 +32,11 @@ class KylinHook(BaseHook):
...
     """
 
+    conn_name_attr = "kylin_conn_id"
+    default_conn_name = "kylin_default"
+    conn_type = "kylin"
+    hook_name = "Apache Kylin"
+
     def __init__(
         self,
         kylin_conn_id: str = "kylin_default",
```
Comment (dirrao) [inline airflow/providers/apache/kylin/hooks/kylin.py:42]: Use default_conn_name here.

**Review by dirrao — CHANGES_REQUESTED** [review]
I have given a one minor comment.

File: airflow/providers/apache/kylin/hooks/kylin.py (hunk 1 above)
Comment (dirrao) [inline airflow/providers/apache/kylin/hooks/kylin.py]:
```suggestion
        kylin_conn_id: str = default_conn_name,
```

Comment (shohamy7) [reply airflow/providers/apache/kylin/hooks/kylin.py]: Thanks for your comment!
Resolved

**Review by dirrao — APPROVED** [review]
LGTM

**Review by Taragolis — APPROVED** [review]

**Review by Taragolis — APPROVED** [review]

**Pull request merged**